atomicapp/providers/kubernetes.py

```diff
@@ -38,227 +44,243 @@ class KubernetesProvider(Provider):
     This class implements deploy, stop and undeploy of an atomicapp on
     Kubernetes provider.
     """
+
+    # Class variables
     key = "kubernetes"
+    namespace = DEFAULT_NAMESPACE
+    k8s_artifacts = {}
+
+    # From the provider configuration
     config_file = None
-    kubectl = None
 
-    def init(self):
-        self.namespace = "default"
+    # Essential provider parameters
+    provider_api = "http://localhost:8080"
+    provider_auth = None
+    provider_tls_verify = None
+    provider_ca = None
 
-        self.k8s_manifests = []
+    def init(self):
+        self.k8s_artifacts = {}
 
         logger.debug("Given config: %s", self.config)
         if self.config.get("namespace"):
             self.namespace = self.config.get("namespace")
 
         logger.info("Using namespace %s", self.namespace)
-        if self.container:
-            self.kubectl = self._find_kubectl(Utils.getRoot())
-            kube_conf_path = "/etc/kubernetes"
-            host_kube_conf_path = Utils.get_real_abspath(kube_conf_path)
-            if not os.path.exists(kube_conf_path) and os.path.exists(host_kube_conf_path):
-                if self.dryrun:
-                    logger.info("DRY-RUN: link %s from %s" % (kube_conf_path, host_kube_conf_path))
-                else:
-                    os.symlink(host_kube_conf_path, kube_conf_path)
-        else:
-            self.kubectl = self._find_kubectl()
+
+        self._process_artifacts()
 
         if not self.dryrun:
```

Comment:
maybe we should just make this: `return if self.dryrun` and then we can unindent the rest of the function?